gitlab/mixins.py

```diff
@@ -73,30 +70,23 @@ def head(
             GitlabAuthenticationError: If authentication is not correct
             GitlabHeadError: If the server cannot perform the request
         """
-        if TYPE_CHECKING:
-            assert self.path is not None
-
         path = self.path
         if id is not None:
             path = f"{path}/{utils.EncodedId(id)}"
 
         return self.gitlab.http_head(path, **kwargs)
 
 
-class GetMixin(HeadMixin, _RestManagerBase):
-    _computed_path: Optional[str]
-    _from_parent_attrs: Dict[str, Any]
-    _obj_cls: Optional[Type[base.RESTObject]]
+class GetMixin(HeadMixin[base.TObjCls]):
     _optional_get_attrs: Tuple[str, ...] = ()
-    _parent: Optional[base.RESTObject]
-    _parent_attrs: Dict[str, Any]
-    _path: Optional[str]
-    gitlab: gitlab.Gitlab
 
     @exc.on_http_error(exc.GitlabGetError)
     def get(
-        self, id: Union[str, int], lazy: bool = False, **kwargs: Any
-    ) -> base.RESTObject:
+        self,
+        id: Union[str, int],
+        lazy: bool = False,
+        **kwargs: Any,
+    ) -> base.TObjCls:
```

Comment:
I must have missed these before. There's a few more unrelated formatting changes (im just nitpickign here because it's a large PR so any change to make the diff smaller helps :sweat_smile: )

```suggestion
        self, id: Union[str, int], lazy: bool = False, **kwargs: Any
    ) -> base.TObjCls:
```